vector/src/main/java/im/vector/app/core/services/CallService.kt

```diff
@@ -177,31 +179,42 @@ class CallService : VectorService() {
         alertManager.postVectorAlert(incomingCallAlert)
         val notification = notificationUtils.buildIncomingCallNotification(
                 call = call,
-                title = opponentMatrixItem?.getBestName() ?: call.mxCall.opponentUserId,
+                title = callInformation.matrixItem?.getBestName() ?: callInformation.opponentUserId,
                 fromBg = fromBg
         )
         if (knownCalls.isEmpty()) {
             startForeground(callId.hashCode(), notification)
         } else {
             notificationManager.notify(callId.hashCode(), notification)
         }
-        knownCalls.add(callId)
+        knownCalls.add(callInformation)
     }
 
     private fun handleCallTerminated(intent: Intent) {
         val callId = intent.getStringExtra(EXTRA_CALL_ID) ?: ""
+        val endCallReason = intent.getSerializableExtra(EXTRA_END_CALL_REASON) as EndCallReason
+        val rejected = intent.getBooleanExtra(EXTRA_END_CALL_REJECTED, false)
         alertManager.cancelAlert(callId)
-        if (!knownCalls.remove(callId)) {
+        val terminatedCall = knownCalls.firstOrNull { it.callId == callId }
+        if (terminatedCall == null) {
             Timber.v("Call terminated for unknown call $callId$")
             handleUnexpectedState(callId)
             return
         }
-        val notification = notificationUtils.buildCallEndedNotification()
-        notificationManager.notify(callId.hashCode(), notification)
+        knownCalls.remove(terminatedCall)
         if (knownCalls.isEmpty()) {
             mediaSession?.isActive = false
             myStopSelf()
         }
+        val wasConnected = connectedCallIds.remove(callId)
+        if (!wasConnected && !terminatedCall.isOutgoing && !rejected && endCallReason != EndCallReason.ANSWERED_ELSEWHERE) {
+            val notification = notificationUtils.buildCallMissedNotification(terminatedCall)
+            notificationManager.cancel(callId.hashCode())
+            notificationManager.notify(MISSED_CALL_TAG, terminatedCall.nativeRoomId.hashCode(), notification)
```

Comment:
At the moment it's not planned to have multiple missed calls for the same room, it might be an improvement later when we will have a proper call history management...

Reply:
Ok, thanks for the change in the string resource